stetho/src/main/java/com/facebook/stetho/inspector/elements/android/ViewDescriptor.java

```diff
@@ -69,7 +145,179 @@ private static String getIdAttribute(View element) {
 
   @Override
   public View getViewForHighlighting(Object element) {
-    return (View)element;
+    return (View) element;
+  }
+
+  @Override
+  protected void onGetStyles(View element, StyleAccumulator styles) {
+
+    for (ViewCSSProperty property : getViewProperties()) {
+      try {
+        getCSSStringFromValue(
+            element,
+            property.getCSSName(),
+            property.getValue(element),
+            property.getAnnotation(),
+            styles);
+      } catch (IllegalAccessException | InvocationTargetException e) {
+        LogUtil.e(e, "failed to get style property " + property.getCSSName() +
+            " of element= " + element.toString());
+      }
+    }
+  }
+
+  private static boolean canIntBeMappedToString(@Nullable ViewDebug.ExportedProperty annotation) {
+    return annotation != null
+        && annotation.mapping() != null
+        && annotation.mapping().length > 0;
+  }
+
+  private static String mapIntToStringUsingAnnotation(
+      int value,
+      @Nullable ViewDebug.ExportedProperty annotation) {
+    if (!canIntBeMappedToString(annotation)) {
+      throw new IllegalStateException("Cannot map using this annotation");
+    }
+
+    for (ViewDebug.IntToString map : annotation.mapping()) {
+      if (map.from() == value) {
+        return map.to();
+      }
+    }
+
+    // no mapping was found even though one was expected ):
+    return NONE_MAPPING;
+  }
+
+  private static boolean canFlagsBeMappedToString(@Nullable ViewDebug.ExportedProperty annotation) {
+    return annotation != null
+        && annotation.flagMapping() != null
+        && annotation.flagMapping().length > 0;
+  }
+
+  private static String mapFlagsToStringUsingAnnotation(
+      int value,
+      @Nullable ViewDebug.ExportedProperty annotation) {
+
+    if (!canFlagsBeMappedToString(annotation)) {
+      throw new IllegalStateException("Cannot map using this annotation");
+    }
+
+    StringBuilder stringBuilder = new StringBuilder();
```

Comment:
We can optimize away the allocation of a `StringBuilder` in the case where `atLeastOneFlag` ends up being false. We can do this by initializing `stringBuilder` to null here, and then lazily instantiating it in the loop below if it's null.